libsql/internal/http/hranaV2/hranaV2.go

```diff
@@ -286,7 +287,119 @@ func sendPipelineRequest(ctx context.Context, msg *hrana.PipelineRequest, url st
 	return result, false, nil
 }
 
+func (h *hranaV2Conn) executeMsg(ctx context.Context, query string, msg *hrana.PipelineRequest) (*hrana.PipelineResponse, error) {
+	result, err := h.sendPipelineRequest(ctx, msg, false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute SQL: %s\n%w", query, err)
+	}
+
+	for _, r := range result.Results {
+		if r.Error != nil {
+			return nil, fmt.Errorf("failed to execute SQL: %s\n%s", query, r.Error.Message)
+		}
+		if r.Response == nil {
+			return nil, fmt.Errorf("failed to execute SQL: %s\n%s", query, "no response received")
+		}
+	}
+	return result, nil
+}
+
+func (h *hranaV2Conn) executeInChunks(ctx context.Context, query string, wantRows bool) (*hrana.PipelineResponse, error) {
+	iterator := sqliteparserutils.CreateStatementIterator(query)
+
+	stmt, _, isEOF := iterator.Next()
+	// Skip all empty statements
+	for {
+		if stmt != "" {
+			break
+		}
+		stmt, _, isEOF = iterator.Next()
+		if isEOF {
+			break
+		}
+	}
+	if isEOF {
+		msg := &hrana.PipelineRequest{}
+		executeStream, err := hrana.ExecuteStream(stmt, nil, wantRows)
```

Comment:
What will we do if `stmt` is empty (so actually query has no non-empty statements)?

Reply:
That behaviour stays unchanged. I haven't checked what it will be as I'm keeping the existing behaviour so no regression there.